Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -295,6 +296,39 @@ String Element::inner_html() const
     return builder.to_string();
 }
 
+void Element::set_inner_text(StringView text)
+{
+    remove_all_children();
+    append_child(document().create_text_node(text));
+
+    set_needs_style_update(true);
+    document().schedule_style_update();
+    document().invalidate_layout();
+}
+
+String Element::inner_text()
+{
+    StringBuilder builder;
+
+    // innerText for element being rendered takes visibility into account, so force a layout and then walk the layout tree.
+    document().layout();
+    if (!layout_node())
+        return text_content();
+
+    Function<void(const LayoutNode&)> recurse = [&](auto& node) {
+        for (auto* child = node.first_child(); child; child = child->next_sibling()) {
+            if (child->is_text())
+                builder.append(downcast<LayoutText>(*child).text_for_rendering());
```

Comment:
`text_for_rendering()` has collapsed whitespace characters, is that really what we want here?
If not, we should get `downcast<LayoutText>(*child).node().data()` instead (the DOM text.)

Reply:
Yes, this should do whitespace collapsing. I tweaked the test to make this explicit.

http://perfectionkills.com/the-poor-misunderstood-innerText/ (which I found just now) has some details if you're curious. Sounds like the property's been standardized since: https://html.spec.whatwg.org/multipage/dom.html#the-innertext-idl-attribute The algorithm there is a lot more detailed than this first draft.

The blog post makes the interesting point that Selection.toString() (which what select->copy implicitly uses, I guess?) should have the same behavior as reading innerText.